src/NumericalAlgorithms/Interpolation/InterpolationTargetApparentHorizon.hpp

```diff
@@ -0,0 +1,188 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "ApparentHorizons/FastFlow.hpp"
+#include "ApparentHorizons/Strahlkorper.hpp"
+#include "ApparentHorizons/Tags.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Informer/Verbosity.hpp"
+#include "NumericalAlgorithms/Interpolation/SendPointsToInterpolator.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+namespace db {
+template <typename TagsList>
+class DataBox;
+}  // namespace db
+namespace intrp {
+namespace Tags {
+template <typename Metavariables>
+struct TemporalIds;
+}  // namespace Tags
+}  // namespace intrp
+namespace OptionTags {
+struct Verbosity;
+}  // namespace OptionTags
+namespace Tags {
+struct Verbosity;
+}  // namespace Tags
+/// \endcond
+
+namespace intrp {
+
+namespace OptionHolders {
+/// Options for finding an apparent horizon.
+template <typename Frame>
+struct ApparentHorizon {
+  /// See Strahlkorper for suboptions.
+  struct InitialGuess {
+    static constexpr OptionString help = {"Initial guess"};
+    using type = Strahlkorper<Frame>;
+  };
+  /// See ::FastFlow for suboptions.
+  struct FastFlow {
+    static constexpr OptionString help = {"FastFlow options"};
+    using type = ::FastFlow;
+  };
+  using options = tmpl::list<InitialGuess, FastFlow, ::OptionTags::Verbosity>;
+  static constexpr OptionString help = {
+      "Provide an initial guess for the apparent horizon surface\n"
+      "(Strahlkorper) and apparent-horizon-finding-algorithm (FastFlow)\n"
+      "options."};
+
+  ApparentHorizon(Strahlkorper<Frame> initial_guess_in, ::FastFlow fast_flow_in,
+                  Verbosity verbosity_in) noexcept;
+
+  ApparentHorizon() = default;
+  ApparentHorizon(const ApparentHorizon& /*rhs*/) = default;
```

Comment:
the copy constructor is also `delete`d in other option-creatable types (e.g. `ScalarWave::Solutions::PlaneWave`, @nilsdeppe could you briefly explain why we do that?)

Reply:
Because creating from options doesn't need to be copyable as far as I'm aware. In the PlaneWave case it holds a `unique_ptr` so it definitely can't be easily copied.

Reply:
Ok, I was able to delete the copy constructor.  This required adding rvalue refs and std::moves in some of the tests.  Since I did this for InterpolationTargetApparentHorizon, I also did the same for all the other InterpolationTargets (and tests thereof that needed it).  I put all the noncopyable-related changes a separate commit; that commit touches many files but does the same thing in all of them.

Reply:
Looks good 👍